python/ray/experimental/sgd/pytorch/pytorch_trainer.py

```diff
@@ -65,8 +66,7 @@ def __init__(self,
         self.optimizer_timer = utils.TimerStat(window_size=1)
 
         if resources_per_replica is None:
-            resources_per_replica = utils.Resources(
-                num_cpus=1, num_gpus=0, resources={})
+            resources_per_replica = Resources(cpu=1, gpu=0)
 
         if backend == "auto":
             backend = "nccl" if resources_per_replica.num_gpus > 0 else "gloo"
```

Comment:
i think this needs to change

Reply:
everywhere using `num_gpus` -> `gpu`

Reply:
actually, hm maybe we should keep it and instead extend `Resources` to use `num_cpus` and `num_gpus`.

Reply:
`int(use_gpu)`